### PR TITLE
MB-7097 Admin API: Add optimistic locking to Webhook Subscriptions

### PR DESCRIPTION
**Pull request by sandy-wright**
## Description

This PR updates the `getWebhookSubscription` and `updateWebhookSubscription` Admin API endpoints to handle optimistic locking/concurrency control. This involved:

* returning an `eTag` value in the response for `getWebhookSubscription`
* adding a required `If-Match` header to the request for `updateWebhookSubscription`
* modifying the payload/model conversion functions to account for the `eTag` value
* passing the eTag into the `UpdateWebhookSubscription` service object
* modifying the `dataProvider` for the Admin UI to set an `If-Match` header for UPDATE requests
* updating tests

## Setup

We don't have an "Update Webhook Subscription" page in the Admin UI just yet, so this can only be manually tested through Postman. Please follow these instructions to set up your testing environment: https://github.com/transcom/mymove/wiki/Test-Admin-and-Office-APIs-with-Postman

Once you're all set up, follow these steps:

* Run your server and client with `make server_run` and `make admin_client_run`.
* Navigate to http://adminlocal:3000/system/webhook_subscriptions
* Pick a webhook subscription from the list. If there are none, create one using the "+ Create" button in the top right of the screen. 
   * The "Subscriber Id" needs to be `5db13bb4-6d29-4bdb-bc81-262f4513ecf6` (our fake Prime ID). The other fields can be whatever you fancy.
* Grab the ID for this webhook subscription and go to Postman. 
* Hit the `getWebhookSubscription` endpoint with your ID. Grab the `eTag` value from the output.
* Go to the `updateWebhookSubscription` endpoint. 
   * Put the ID in the url and the `eTag` in the "If-Match" header. You'll also need the "X-CSRF-Token" here (refer to https://github.com/transcom/mymove/wiki/Test-Admin-and-Office-APIs-with-Postman for more info).
   * Send a request body like:
```json
{
  "callbackUrl": "new.url.here"
}
```
* Verify that you got a 200 response code and that you see a new `eTag` in the response body.
* Send it again, with all the same info! This time you should see a 412 response code indicating that the `eTag` value was stale.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7097) for this change.

### CODE REVIEW

File: pkg/handlers/adminapi/payloads/model_to_payload.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"github.com/go-openapi/strfmt"
+
+	"github.com/transcom/mymove/pkg/etag"
+	"github.com/transcom/mymove/pkg/gen/adminmessages"
+	"github.com/transcom/mymove/pkg/handlers"
+	"github.com/transcom/mymove/pkg/models"
+)
+
+// WebhookSubscriptionPayload converts a webhook subscription payload to a model
```
Comment (shimonacarvalho) [inline pkg/handlers/adminapi/payloads/model_to_payload.go:12]: Looks like it converts model to payload.

File: pkg/handlers/adminapi/payloads/payload_to_model.go [hunk 2]
```diff
@@ -57,3 +56,16 @@ func WebhookSubscriptionModel(sub *adminmessages.WebhookSubscription) *models.We
 
 	return model
 }
+
+// WebhookSubscriptionModelFromCreate converts a payload for creating a webhook subscription to a model
+func WebhookSubscriptionModelFromCreate(sub *adminmessages.CreateWebhookSubscription) *models.WebhookSubscription {
+	model := &models.WebhookSubscription{
+		EventKey:     *sub.EventKey,
+		CallbackURL:  *sub.CallbackURL,
+		SubscriberID: uuid.FromStringOrNil(sub.SubscriberID.String()),
```
Comment (shimonacarvalho) [inline pkg/handlers/adminapi/payloads/payload_to_model.go:65]: Minor nitpick - Could you put a comment on these 2 dereferences noting that these fields are required in the swagger?
Just helps others learn they can't just dereference like this unless they check that. Thanks!

File: pkg/handlers/adminapi/webhook_subscriptions.go [hunk 3]
```diff
@@ -16,38 +16,6 @@ import (
 	"github.com/transcom/mymove/pkg/services/query"
 )
 
-// payloadForWebhookSubscriptionModel converts from model to payload
```
Comment (shimonacarvalho) [inline pkg/handlers/adminapi/webhook_subscriptions.go:19]: Ooh thanks for moving these to the right place!

**Review by shimonacarvalho — APPROVED** [review]
All the backend stuff looks good to me! Maybe get a second pair of eyes for the FE? Just some minor comment related stuff that's not a blocker. 

Oh I see you need a manual test. I will do that tomorrow!

**Comment by shimonacarvalho** [comment]
Okay tested and works.

File: pkg/handlers/adminapi/payloads/model_to_payload.go (hunk 1 above)
Comment (sandy-wright) [reply pkg/handlers/adminapi/payloads/model_to_payload.go:12]: whoops! Nice catch!

**Pull request merged**